src/main/java/address/browser/HyperBrowser.java

```diff
@@ -74,7 +76,8 @@ private void initialiseHyperBrowser(){
         for (int i = 0; i < noOfPages; i++){
             EmbeddedBrowser browser;
             if (browserType == FULL_FEATURE_BROWSER){
-                browser = new JxBrowserAdapter(new Browser());
+                BrowserContext context = new BrowserContext(new BrowserContextParams("tmpTab" + i));
```

Comment:
I agree, but previously, there is a dead lock on jxBrowser getUrl method when this is not set. After I changed this, the problem seems to be gone.